app/models/resource.rb

```diff
@@ -20,6 +21,19 @@ class Resource < ApplicationRecord
 
   validates :title, :resource_type, presence: true
 
-  after_save { SearchIndex.add(self) }
-  after_destroy { SearchIndex.remove(self) }
+  after_save do
+    SearchIndex.new(
+      model_name: self.class.name,
+      id: id,
+      async: true,
+    ).add
+  end
+
+  after_destroy do
+    SearchIndex.new(
```

Comment:
Would it make sense to have `RemoveFromIndexJob` directly here instead of having the path go `SearchIndex` -> `RemoveFromIndexJob` -> `SearchIndex`?

Not saying my suggestion is better -- but it feels a bit confusing to have one class refer to another, no? Alternatively, would it make sense to have the code to actually add or remove the item from the index to exist within `AddToIndexJob` and `RemoveFromIndexJob`? Curious what you think is best for this part.